### PR TITLE
preparing beta release

### PR DESCRIPTION
**Pull request by DavertMik**
Updated changelog & docs
Preparing release.

### CODE REVIEW

File: docs/pageobjects.md [hunk 1]
```diff
@@ -30,11 +30,8 @@ Required objects can be obtained via parameters in tests or via global `inject()
 const { I, myPage, mySteps } = inject();
 
 // inject objects for a test by name
-Scenario('sample test', (I, myPage, mySteps) => {
-  // ...
-})
-```
-
+Scenario('sample test', ({ I, myPage, mySteps }) => {
+  // ...\w+
```
Comment (DavertMik) [inline docs/pageobjects.md:34]:
```suggestion
  // ...
  }
```

File: docs/puppeteer.md [hunk 2]
```diff
@@ -259,12 +259,23 @@ I.usePuppeteerTo('emulate offline mode', async ({ page, browser }) => {
...
 
-Because all Puppeteer commands are asynchronous callback function must be async.
+> Puppeteer commands are asynchronous so a callback function must be async.
+
+A Puppeteer helper is passed as argument for callback, so you can combine Puppeteer API with CodeceptJS API:
+
+```js
+I.usePuppeteerTo('emulate offline mode', async (Puppeteer) => {
+  // access internal objects browser, page, context of helper
+  await Puppeteer,page.setOfflineMode(true);
```
Comment (pablopaul) [inline docs/puppeteer.md:269]:
```suggestion
  await Puppeteer.page.setOfflineMode(true);
```

File: docs/pageobjects.md [hunk 3]
```diff
@@ -108,9 +105,11 @@ You can include this pageobject in test by its name (defined in `codecept.json`)
 it should be added to list of test arguments to be included in test:
 
 ```js
-Scenario('login', (I, loginPage) => {
+Scenario('login', ({ I, loginPage }) => {
+
```
Comment (pablopaul) [inline docs/pageobjects.md:109]:
```suggestion
```

File: docs/pageobjects.md [hunk 4]
```diff
@@ -108,9 +105,11 @@ You can include this pageobject in test by its name (defined in `codecept.json`)
...
   loginPage.sendForm('john@doe.com','123456');
   I.see('Hello, John');
+
```
Comment (pablopaul) [inline docs/pageobjects.md:112]:
```suggestion
```

File: docs/playwright.md [hunk 5]
```diff
@@ -206,7 +206,7 @@ CodeceptJS allows you to implement custom actions like `I.createTodo` or use **P
 TO launch additional browser context (or incognito window) use `session` command.
 
 ```js
-Scenario('I try to open this site as anonymous user', () => {
+Scenario('I try to open this site as anonymous user', ({  }) => {
```
Comment (pablopaul) [inline docs/playwright.md:209]:
```suggestion
Scenario('I try to open this site as anonymous user', () => {
```

File: docs/playwright.md [hunk 6]
```diff
@@ -253,7 +253,7 @@ To enable device emulation for a specific test, create an additional browser ses
 ```js
 const { devices } = require('playwright');
 
-Scenario('website looks nice on iPhone', () => {
+Scenario('website looks nice on iPhone', ({  }) => {
```
Comment (pablopaul) [inline docs/playwright.md:256]:
```suggestion
Scenario('website looks nice on iPhone', () => {
```

File: docs/plugins.md [hunk 7]
```diff
@@ -782,6 +782,56 @@ Possible config options:
 
 -   `config` **any** 
 
+## tryTo
+
+Adds global `tryTo` function inside of which all failed steps won't fail a test but will return true/false.
+
+```js
+const result = await tryTo(() => I.see('Welcome'));
```
Comment (pablopaul) [inline docs/plugins.md:790]: @DavertMik syntax `I.tryTo( () => see("welcome")` would be even a bit nicer :)

File: docs/plugins.md [hunk 8]
```diff
@@ -782,6 +782,56 @@ Possible config options:
...
+
+// if user is on page, result => true
+// if user is on page, result => false
+```
+
+Disables retryFailedStep plugin for steps inside a block;
```
Comment (pablopaul) [inline docs/plugins.md:796]:
```suggestion
Disables retryFailedStep plugin for steps inside a block.
```

Comment (pablopaul) [reply docs/plugins.md:796]: @DavertMik Is this info about that retryFailedStep plugin is disabled for steps inside "tryTo"?

File: docs/plugins.md [hunk 9]
```diff
@@ -782,6 +782,56 @@ Possible config options:
...
+
+Use this plugin if:
+
+-   you need to perform multiple assertions inside a test
+-   there is A/B testing on a website you test
+-   there is "Accept Cookie" banner which may surprisingly appear on a page.
```
Comment (pablopaul) [inline docs/plugins.md:802]:
```suggestion
-   there is "accept cookie" banner which may surprisingly appear on a page
```

File: docs/index.md [hunk 10]
```diff
@@ -14,7 +14,7 @@ Type in commands to complete the test scenario.
 Successful commands will be saved into a file.
 
 ```js
-Scenario('Checkout test', () => {
+Scenario('Checkout test', ({  }) => {
```
Comment (DavertMik) [inline docs/index.md:17]:
```suggestion
Scenario('Checkout test', ({ I }) => {
```

File: docs/index.md [hunk 11]
```diff
@@ -68,7 +68,7 @@ Features:
 
 Each executed step will be printed on screen when running with `--steps`
 ```js
-Scenario('Checkout test', () => {
+Scenario('Checkout test', ({  }) => {
```
Comment (DavertMik) [inline docs/index.md:71]:
```suggestion
Scenario('Checkout test', ({ I }) => {
```

File: docs/plugins.md (hunk 7 above)
Comment (DavertMik) [reply docs/plugins.md:790]: nope, from JavaScript point of view, it is not :)

File: docs/plugins.md (hunk 8 above)
Comment (DavertMik) [reply docs/plugins.md:796]: Yes, you are right! Can you propose better wording here?

**Pull request merged**